src/OSSupport/ServerHandleImpl.cpp

```diff
@@ -63,10 +63,15 @@ cServerHandleImpl::~cServerHandleImpl()
 void cServerHandleImpl::Close(void)
 {
 	// Stop the listener sockets:
-	evconnlistener_disable(m_ConnListener);
+	if (m_ConnListener != nullptr)
+	{
+		evconnlistener_disable(m_ConnListener);
+		m_ConnListener = nullptr;
```

Comment:
This is wrong, it causes the socket to not be closed when the server is finally destroyed.